integration-tests/worker/test/autoinstall.test.ts

```diff
@@ -0,0 +1,60 @@
+// stress test for autoinstall
+// this could evolve  into stress testing, benchmarking or artillery generally?
+// Also I may skip this in CI after the issue is fixed
+
+import test from 'ava';
+import path from 'node:path';
+
+import { initLightning, initWorker } from '../src/init';
+import { createAttempt, createJob } from '../src/factories';
+
+const generate = (adaptor, version) => {
+  const specifier = `@openfn/language-${adaptor}@${version}`;
+  const job = createJob({
+    body: `fn(() => ({ data: "${adaptor}" }))`,
+    adaptor: specifier,
+  });
+  return createAttempt([], [job], []);
+};
+
+let lightning;
+let worker;
+
+const run = async (attempt) => {
+  return new Promise<any>(async (done, reject) => {
+    lightning.on('attempt:complete', (evt) => {
+      if (attempt.id === evt.attemptId) {
+        done(lightning.getResult(attempt.id));
+      }
+    });
+
+    lightning.enqueueAttempt(attempt);
+  });
+};
+
+test.before(async () => {
+  const lightningPort = 4321;
+
+  lightning = initLightning(lightningPort);
+
+  ({ worker } = await initWorker(lightningPort, {
+    repoDir: path.resolve('tmp/repo/autoinstall'),
+  }));
+});
+
+test.after(async () => {
+  lightning.destroy();
+  await worker.destroy();
+});
+
+test('autoinstall three things at once', async (t) => {
```

Comment:
This test  regularly fails against main - on this branch it's stable. I think this is really encouraging.

I think I'll soon move this test (see the comments at the top) but right now I'm happy to leave it - I'm more interested in getting the fix deployed.